src/Client/CurlHandler.php

```diff
@@ -57,8 +57,25 @@ public function __destruct()
         }
     }
 
+    /**
+     * @param array $request
+     *
+     * @return CompletedFutureArray
+     */
     public function __invoke(array $request)
     {
+        return new CompletedFutureArray(
+            $this->invokeAsArray($request)
+        );
+    }
+
+    /**
+     * @param array $request
+     *
+     * @return array
+     */
+    public function invokeAsArray(array $request)
```

Comment:
can you make this private? I don't think this needs to be part of the public API.